content/2023-12-28-embedded-hal-v1.md

```diff
@@ -0,0 +1,70 @@
+such as GPIO, UART, SPI or I2C. They allow writing drivers (for sensors, displays, actuators, network adapters...) in
+a generic way, so they work out of the box on any microcontroller with an `embedded-hal` implementation, without
+modifying them. It's a central piece of the Embedded Rust ecosystem, ensuring interoperability throughout.
+
+The 1.0 release has been in the works for a while, since 2020. Now that it's out, we consider all
+traits in it to be *stable*. The plan is to extend `embedded-hal` with more traits in future 1.x releases,
+not doing more breaking changes (i.e. there are no plans for a a 2.0 release). This will provide a stable
+base for building HALs and drivers.
+
+So, what's new in `embedded-hal` 1.0?
+
+<!-- more -->
+
+## Focus on drivers
+
+Previous versions of `embedded-hal` had a dual goal of standardizing HAL APIs for end users, and allowing writing generic drivers. 
+With time, we've seen both goals sometimes conflict, and the latter one brings much more value, so 1.0 focuses on just that.
+
+We've simplified some traits and merged others to remove interoperability gotchas.
+
+We've removed traits that were found to not be usable for generic drivers (most notably timers). The plan is to add
+them back in the future, with a better suited design. See the migration guide for details and links to the tracking issues.
+
+## Async
+
+A new addition is the `embedded-hal-async` crate, containing async versions of the traits. With today's Rust 1.75 release,
+async traits are available on Rust stable. They can be used with no heap allocations and no dynamic
+dispatch (unlike previous macro-based polyfills like the `async-trait` crate), so they are a great fit for bare-metal embedded usage.
+
+Most `embedded-hal-async` traits are async versions of their blocking counterparts.
+
+However, one highlight is the [`digital::Wait`](https://docs.rs/embedded-hal-async/1.0.0/embedded_hal_async/digital/trait.Wait.html), with methods like `wait_for_high()` and `wait_for_low()`. This trait adds support for using "IRQ" GPIO pins typically used by SPI and I2C devices to send an interrupt to the microcontroller. This has been a frequently requested feature, and has turned out to be hard to abstract with traits, until now that we have async traits in Rust.
+
+## SPI bus sharing
+
+![SPI bus/device diagram](../spi-shared-bus.svg)
+
+The [`SpiDevice`](https://docs.rs/embedded-hal/1.0.0/embedded_hal/spi/index.html) trait now allows sharing a SPI bus between multiple devices, each selected with its own CS pin. The design allows for unrelated drivers to talk to different devices in the same bus, without conflicts, and without being aware of each other. 
+
+The trait is agnostic to the kind of mutex/locking mechanism. The [`embedded-hal-bus`](https://docs.rs/embedded-hal-bus/0.1.0/embedded_hal_bus/spi/index.html) crate provides implementations for commonly used mutexes, but it is possible to write your own for e.g. the mutex of your choice RTOS.
```

Comment:
“agnostic to” sounds a bit odd to me, I’d write “agnostic about”. Also there’s either an “of” missing or something else weird about “the mutex of your choice RTOS.”